File: backend/usport/src/main/java/com/app/usport/usermetric/UserMetricService.java
package com.app.usport.usermetric;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;


@Service
public class UserMetricService {
    private UserMetricRepository userMetricRepository;

    @Autowired
    public UserMetricService(UserMetricRepository userMetricRepository) {
        this.userMetricRepository = userMetricRepository;
    }

    UserMetric getUserMetric(String userId) {
        return this.userMetricRepository.getUserMetric(userId);
    }

    public UserMetric createUserMetric(UserMetric userMetric) {
        return this.userMetricRepository.createUserMetric(userMetric);
    }

    public List<UserMetric> getTopUserMetrics() {
        return userMetricRepository.getTopUserMetrics();
    }
}

File: frontend/usport/src/services/UserMetricService.js
import axios from 'axios';

class UserMetricService{

    getUserMetric(userId){
        return axios.get('http://localhost:8080/api/user_metric/' + userId);
    }

    createUserMetric(requestBody){
        return axios.post('http://localhost:8080/api/user_metric/create', requestBody);
    }
    
}

const userMetricService = new UserMetricService();

export default userMetricService;
